Backend/myfitapp/views.py

```diff
@@ -0,0 +1,467 @@
+            },
+        )
+
+        if created and not body_metrics.weight:
+            raise ValidationError("Weight is required for new entries.")
+
+        return body_metrics
+
+
+# View to search a food to check calories
+class SearchFoodView(APIView):
+    """
+    This view will use an API to get the nutritional information
+    like calories, fats, protein, sugars, sodium things like that,
+    about the food
+    """
+
+    permission_classes = [IsAuthenticated]
+
+    def post(self, request):
+        input_serializer = SearchSerializer(data=request.data)
+        if input_serializer.is_valid():
+            food_name = input_serializer.validated_data["food_name"].title()
+            quantity = input_serializer.validated_data["quantity"]
+
+            return Response(nutritional_facts(food_name, quantity))
+        raise ValidationError(detail="Not Valid")
+
+
+# View to search a recipe
+class SearchRecipe(APIView):
+    """
+    This View uses an API to give the recipe of the dish
+    """
+
+    permission_classes = [IsAuthenticated]
+
+    def post(self, request):
+        input_serializer = RecipeSerializer(data=request.data)
+        if input_serializer.is_valid():
+            food_name = input_serializer.validated_data["food_name"]
+
+            return Response(recipe_recommendation(food_name))
+        raise ValidationError(detail="Not Valid")
+
+
+# View to list all the foods logged that day
+class ListFoodsView(APIView):
+    """
+    The user can see all the foods they consumed on that day
+    """
+
+    permission_classes = [IsAuthenticated]
+
+    def get(self, request):
+        query_date = request.query_params.get("date", timezone.now().date())
+
+        try:
+            query_date = timezone.now().date().fromisoformat(str(query_date))
+        except ValueError:
+            return Response(
+                {"error": "Invalid date format. Use YYYY-MM-DD."}, status=400
+            )
+
+        foods = UserFoodLog.objects.filter(
+            profile=request.user.profile, date=query_date
+        )
+        serializer = UserFoodLogSerializer(foods, many=True)
+
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+# View to track the user calories
+class UserFoodLogView(APIView):
+    """
+    View to log the food, it also shows how much calorie is remaining
+    """
+
+    permission_classes = [IsAuthenticated]
+
+    def post(self, request):
+        input_serializer = UserFoodLogSerializer(data=request.data)
+        try:
+            profile = Profile.objects.get(user=request.user)
+        except Profile.DoesNotExist:
+            return Response(
+                {"error": "Profile not found for the user."},
+                status=status.HTTP_404_NOT_FOUND,
+            )
+
+        if input_serializer.is_valid():
+            input_serializer.save(profile=profile)
+
+            daily_summary = UserFoodLog.daily_summary(profile)
+
+            return Response(
```

Comment:
use serializers for response.

Reply:
did this for customized response.
changed to serializers now.